### PR TITLE
Make filtering by actions/events in sessions possible

### PR DESCRIPTION
**Pull request by macobo**
## Changes

UI is pretty ugly, however functional:

Screenshot from 2020-12-17 16-20-10

Also did some refactoring and typescript cleanup along the way.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/lib/components/SelectBox.tsx [hunk 1]
```diff
@@ -74,15 +74,15 @@ export function SelectBox({
                         style={{ width: '100%', borderRadius: 0 }}
                     />
                     <div className="search-list">
-                        {items.map((item) => (
-                            <>
+                        {items.map((item, index) => (
+                            <Fragment key={index}>
```
Comment (paolodamico) [inline frontend/src/lib/components/SelectBox.tsx:78]: **non-actionable** ❤️! one less error in the console when developing

File: frontend/src/scenes/sessions/SessionsTable.tsx [hunk 2]
```diff
@@ -187,7 +188,11 @@ export function SessionsTable({ personIds, isPersonPage = false }: SessionsTable
                 <Button onClick={nextDay} icon={<CaretRightOutlined />} />
             </Space>
 
-            {featureFlags['filter_by_session_props'] && (
+            {featureFlags['filter_by_session_props'] && user?.is_multi_tenancy && (
+                <SessionActionFilters actionFilter={actionFilter} updateActionFilter={updateActionFilter} />
+            )}
+
+            {featureFlags['filter_by_session_props'] && user?.is_multi_tenancy && (
```
Comment (paolodamico) [inline frontend/src/scenes/sessions/SessionsTable.tsx:195]: Nit: maybe put both filters under the same check? We're checking the exact same conditions

**Review by paolodamico — APPROVED** [review]
Great stuff! We can definitely improve UI for this

UI/UX low hanging fruit improvements:
- Having both filter options looks ugly and is a bit confusing, maybe remove one of them?
    
- Make the dropdown the same length as the filter above and below.
- Move one of the filters (thinking the recording duration) to be next to the date selector, instead of the three of them in a column.

File: frontend/src/scenes/sessions/SessionsTable.tsx (hunk 2 above)
Comment (macobo) [reply frontend/src/scenes/sessions/SessionsTable.tsx]: Oops. Fixed.

**Comment by macobo** [comment]
Re UX feedback - agreed in broad strokes that this needs work. Honestly I'd love to scrap everything we have currently UI-wise and redesign this from scratch.

As-is, I'm proposing keeping this behind a feature flag until we do - unless you think with the small fixes it's releasable as-is?

>Having both filter options looks ugly and is a bit confusing, maybe remove one of them?

Problem: You want to filter user properties separately from events/actions (e.g. show sessions from a certain cohort).

I tried giving the two different labels, but `Add user property filter` felt really clunky.

**Pull request merged**